src/core/managers/TransactionManager.ts

```diff
@@ -0,0 +1,149 @@
+import { Transaction } from '@multiversx/sdk-core/out';
+import axios, { AxiosError } from 'axios';
+import { BATCH_TRANSACTIONS_ID_SEPARATOR } from 'constants/transactions.constants';
+import { getAccount } from 'core/methods/account/getAccount';
+import { networkSelector } from 'store/selectors';
+import { getState } from 'store/store';
+import { GuardianActionsEnum } from 'types';
+import { BatchTransactionsResponseType } from 'types/serverTransactions.types';
+import { SignedTransactionType } from 'types/transactions.types';
+
+export class TransactionManager {
+  private static instance: TransactionManager | null = null;
+
+  private constructor() {}
+
+  public static getInstance(): TransactionManager {
```

Comment:
is it needed?

Reply:
yes, because the constructor is private and this is the only way to get an instance